CollectPoints/qSlicerCollectPointsModuleWidget.cxx

```diff
@@ -263,26 +263,17 @@ void qSlicerCollectPointsModuleWidget::onAnchorTransformNodeSelected()
 }
 
 //-----------------------------------------------------------------------------
-void qSlicerCollectPointsModuleWidget::onOutputNodeAdded( vtkMRMLNode* newNode )
+void qSlicerCollectPointsModuleWidget::onOutputNodeAdded( vtkMRMLNode* vtkNotUsed(newNode) )
 {
   Q_D( qSlicerCollectPointsModuleWidget );
 
-  vtkMRMLMarkupsNode* outputMarkupsNode = vtkMRMLMarkupsNode::SafeDownCast( newNode );
-  if ( outputMarkupsNode != NULL )
-  {
-    outputMarkupsNode->CreateDefaultDisplayNodes();
-  }
-
-  vtkMRMLModelNode* outputModelNode = vtkMRMLModelNode::SafeDownCast( newNode );
-  if ( outputModelNode != NULL )
+  vtkMRMLCollectPointsNode* collectPointsNode = vtkMRMLCollectPointsNode::SafeDownCast( d->ParameterNodeComboBox->currentNode() );
+  if ( collectPointsNode == NULL )
   {
-    // display node should be set to show points only
-    outputModelNode->CreateDefaultDisplayNodes();
-    vtkMRMLModelDisplayNode* outputModelDisplayNode = outputModelNode->GetModelDisplayNode();
-    outputModelDisplayNode->SetRepresentation( vtkMRMLDisplayNode::PointsRepresentation );
-    const int DEFAULT_POINT_SIZE = 5; // This could be made an advanced display setting in the GUI
-    outputModelDisplayNode->SetPointSize( DEFAULT_POINT_SIZE );
+    qCritical() << Q_FUNC_INFO << ": invalid parameter node";
+    return;
   }
+  collectPointsNode->CreateDefaultDisplayNodesForOutputNode();
 
   this->updateGUIFromMRML();
```

Comment:
I assume you refer to updateGUIFromMRML? I always do this just in case something at the node level necessitates change in some of the attributes/references.

It also gives a cue when something in the node isn't updating the way it is supposed to (say I choose an output node and for some reason it isn't actually set, then I'll see this in the GUI right away. I won't need to assume that the node I see in the box has been set successfully, I will know for sure one way or the other.)

Reply:
>  I always do this just in case something at the node level necessitates change in some of the attributes/references.

Node changes are already observed (if not, then it's an error that has to be fixed). So, any extra updateGUIFromMRML calls are unnecessary. The computation time is negligible, but the main issue is that it may mask the error of missing node observations.

Reply:
I see. I think most of the modules I've worked with have called updateGUIFromMRML (or the equivalent) manually. I'd been following those examples. What is the proper mechanism for updating the GUI?

Reply:
When you select a node, you add an observer to modification events: 

https://github.com/SlicerIGT/SlicerIGT/blob/8d7fbdc08d39ff6a14f9f38746c0fce7c88444e9/ToolWatchdog/qSlicerWatchdogModuleWidget.cxx#L226-L233

https://github.com/SlicerIGT/SlicerIGT/blob/8d7fbdc08d39ff6a14f9f38746c0fce7c88444e9/FiducialRegistrationWizard/qSlicerFiducialRegistrationWizardModuleWidget.cxx#L488-L495

Reply:
Thanks, I will make sure that the modules I've worked with recently follow those two examples.